plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -648,4 +657,26 @@ public List<Setting<?>> getSettings() {
             );
         return settings;
     }
+
+    @Override
+    public List<SearchPlugin.SearchExtSpec<?>> getSearchExts() {
+        return List
+            .of(
+                new SearchPlugin.SearchExtSpec<>(
+                    GenerativeQAParamExtBuilder.PARAMETER_NAME,
+                    input -> new GenerativeQAParamExtBuilder(input),
+                    parser -> GenerativeQAParamExtBuilder.parse(parser)
+                )
+            );
+    }
+
+    @Override
+    public Map<String, Processor.Factory<SearchRequestProcessor>> getRequestProcessors(Parameters parameters) {
+        return Map.of(GenerativeQAProcessorConstants.REQUEST_PROCESSOR_TYPE, new GenerativeQARequestProcessor.Factory());
+    }
+
+    @Override
+    public Map<String, Processor.Factory<SearchResponseProcessor>> getResponseProcessors(Parameters parameters) {
+        return Map.of(GenerativeQAProcessorConstants.RESPONSE_PROCESSOR_TYPE, new GenerativeQAResponseProcessor.Factory(this.client));
+    }
```

Comment:
@msfroh We want to release these behind a feature flag for v2.10.0.  I am thinking of returning empty Lists/Maps if the flag is false.  What do you think?  Is there a cleaner way to handle this?

Reply:
I think that's a very clean way of handling it.

In particular, feature flags are loaded on startup and the set of available processors are loaded on startup. Saying that the processors are available if and only if the feature flag is set sounds correct to me.